cluster-autoscaler/proposals/cp_agnostic_ca.md

```diff
@@ -0,0 +1,158 @@
+# Decoupling ClusterAutoScalerCore (separate library) from CloudProvider implementations
+##### Author: bhargav-naik
+
+### Introduction
+Current ClusterAutoscaler is released as a bundle which includes core and all the CloudProviders included.
+When someone intends to write implementation of the CloudProvider they need to fork the intended version ClusterAutoscaler and provide their implementation.
+In this proposal, a different design approach is presented which will decouple the ClusterAutoscaler core from CloudProviders and make ClusterAutoScalerCore available as a separate release. 
```

Comment:
I don't think this is true anymore based on Version 2

Reply:
@hectorj2f , thanks for the input. I will clean up the sections.

Reply:
done.